gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -1428,3 +1428,91 @@ gst_tensor_meta_info_append_header (GstTensorMetaInfo * meta, GstMemory * mem)
   gst_memory_unmap (new_mem, &new_map);
   return new_mem;
 }
+
+#define NNS_TENSOR_EXTRA_MAGIC 0xf00dc0de
+
+/**
+ * @brief TBU
+*/
+void
+gst_tensors_extra_init (GstTensorExtraInfo * extra)
+{
+  guint i;
+
+  extra->magic = NNS_TENSOR_EXTRA_MAGIC;
+  extra->version = 0;
+  extra->num_extra_tensors = 0;
+  for (i = 0; i < NNS_TENSOR_SIZE_EXTRA_LIMIT; ++i) {
+    gst_tensor_info_init (&extra->infos[i]);
+  }
+}
+
+/**
+ * @brief TBU
+*/
+GstMemory *
+gst_tensors_get_nth_memory (GstBuffer * buffer, guint index)
```

Comment:
Plz consider the param `GstTensorsInfo * info` for the case if num-tensors is 16,
```
gst_tensor_get_nth_memory (GstBuffer * buffer, const GstTensorsInfo * info, const guint index)
{
...
  if (info->num_tensors <= NNS_TENSOR_SIZE_LIMIT)
    return gst_buffer_get_memory (buffer, index);
...
}
```

Reply:
I updated the code, PTAL